dbt/adapters/trino/connections.py

```diff
@@ -119,6 +119,8 @@ def _escape_value(cls, value):
         if value is None:
             return "NULL"
         elif isinstance(value, str):
+            if re.match(r"\w+\s+'.*'", value) is not None:
```

Comment:
This change is inherently linked to this fix, eg. we can't add quotes around DECIMAL '10.9'. In other SQL engines generally you don't have to type the values you insert into a table, so it would have just worked. That's the underlying issue here. I will definitely add a comment here. Maybe worth a trino bug as well if not yet existing.

Reply:
```
trino> use memory.default;
USE
trino:default> create table test_decimal(long_decimal decimal(25,20));
CREATE TABLE
trino:default> insert into test_decimal values (DECIMAL '12345.12345678901234567890');
INSERT: 1 row

Query 20220211_050323_00006_wcrqk, FINISHED, 1 node
Splits: 19 total, 19 done (100.00%)
0.56 [0 rows, 0B] [0 rows/s, 0B/s]

trino:default> select * from test_decimal;
        long_decimal        
----------------------------
 12345.12345678901234567890 
(1 row)
```

Reply:
I think that this check is made at a too low level.

Check `dbt.adapters.trino.connections.ConnectionWrapper.execute`

If you run `dbt seed` in debug, you can see that the bindings come already filled here:

```
bindings = tuple(self._escape_value(b) for b in bindings)
```

Here is one example for the string types:

```
['test', 'abc', "CHAR 'd'", "CHAR 'ghi'", None, None, None, None]
```

I would argue that we shouldn't land in the `_escape_value` method when dealing with bindings which are already handled by the adapter.

Reply:
Good point, we actually do the bindings ourselves in the macro, so no need to go through the _escape_values. Anyway the Python typing system doesn't support all necessary types to fully support all Trino data types. Ideally at some point we shouldn't need to use _escape_value or do this binding here. Trino already knows about the type of the columns and should just accept strings and cast them accordingly similar like how postgres/snowflake do.

Reply:
```
varchar_example,varchar_n_example,char_example,char_n_example
CHAR 'test',abc,d,ghi
```

leads now to inserting `'test'` into the `string_type` table - which is obviously wrong.

Reply:
My plan is to only add the agate detected types as bindings, the other (overriden) columns i will just embed in the sql statement in the seed. Then we can rid ourselves of the regex check. Commit will follow later tonight or tomorrow.

Reply:
Please keep in mind the following:

- `agate` supports just a handful of types: `boolean`, `date`, `date_time`, `number`, `text`, `time_delta`.
- dbt uses the types specified for the seed columns in the `CREATE TABLE` statement.
- if agate doesn't understand a specific type (quite likely because the names it is using are not the same as the SQL type names) defaults to text (string)


I tend to think you are on a good path with overriding `load_csv_rows`.

It is important however that the `bindings` get to their final state before the call

```
      {% do adapter.add_query(sql, bindings=bindings, abridge_sql_log=True) %}
```

In this direction, can we drop `dbt.adapters.trino.connections.ConnectionWrapper._escape_value` ?

Reply:
Yeah, i like that idea, it seems out of place there, also trino-python-client seems to natively support some bindings. So we could make use of that and get rid of _escape_values all together. See following tests:

https://github.com/trinodb/trino-python-client/blob/65506e8f1c90ea8908576d43037be46f53c8604a/tests/integration/test_dbapi_integration.py#L134-L135

I actually doubt that dbt inserts anything with bindings outside of seeds, but definitely converting python types into something trino understands, should be done in trino-python-client.

Reply:
agate uses decimals as number data type. Those are however not suported in trino-python-client. So for now we can't rid ourselves of the _escape_values. Also jinja lacks some python features to do the tests currently done in _escape_values. So I left that code as is, I will be looking next at improving the typings in the trino-python-client.